src/main.rs

```diff
@@ -62,22 +61,110 @@ fn main() -> anyhow::Result<()> {
+                && x.z < avg.z + cutoff * std_dev.z
+        })
+        .copied()
+        .collect::<Vec<DVec3>>();
+
+    let n = positions.len();
+    let n_filtered = positions_filtered.len();
+    let avg_filtered = positions_filtered.iter().copied().sum::<DVec3>() / n_filtered as f64;
+    let std_dev_filtered = (positions_filtered
+        .iter()
+        .copied()
+        .map(|r| (r - avg_filtered).powf(2.))
+        .sum::<DVec3>()
+        / (n_filtered - 1) as f64)
+        .powf(0.5);
+    let std_dev_m = {
+        let (y, x, z) = geodetic2enu(
+            (avg_filtered + std_dev_filtered).x,
+            (avg_filtered + std_dev_filtered).y,
+            (avg_filtered + std_dev_filtered).z,
+            avg_filtered.x,
+            avg_filtered.y,
+            avg_filtered.z,
+            map_3d::Ellipsoid::WGS84,
+        );
+        DVec3::from((x, y, z))
+    };
+
     if short {
-        println!("{}, {}, {}", avg.x, avg.y, avg.z);
+        println!("{}, {}, {}", avg_filtered.x, avg_filtered.y, avg_filtered.z);
     } else {
-        let formatted = format!("({:.4}º, {:.4}º, {:.1}m)", avg.x, avg.y, avg.z).bold();
-        let formatted_raw = format!("({}, {}, {})", avg.x, avg.y, avg.z).italic();
-        println!("Average: {formatted} {formatted_raw}\n");
-
-        println!("Number of entries: {n}");
-        let formatted = format!("({:.6}º, {:.6}º, {:.3}m)", std_dev.x, std_dev.y, std_dev.z);
-        let formatted_m = format!(
-            "Horizontally: ~({:.2}m, {:.2}m)",
-            std_dev.x * EARTH_RADIUS,
-            std_dev.y * EARTH_RADIUS
+        let formatted = format!(
+            "({:.4}º, {:.4}º, {:.1}m)",
+            avg_filtered.x, avg_filtered.y, avg_filtered.z
+        )
+        .bold();
+        let formatted_raw = format!(
+            "({}, {}, {})",
+            avg_filtered.x, avg_filtered.y, avg_filtered.z
         )
         .italic();
+        println!("Average: {formatted} {formatted_raw}\n");
+
+        let formatted = format!("({} after filter)", n_filtered).italic();
+        println!("Number of entries: {n} {}", formatted);
+        let formatted = format!(
+            "({:.6}º, {:.6}º, {:.3}m)",
+            std_dev_filtered.x, std_dev_filtered.y, std_dev_filtered.z
+        );
+        let formatted_m =
+            format!("Horizontally: ~({:.2}m, {:.2}m)", std_dev_m.x, std_dev_m.y).italic();
         println!("Standard deviation: {formatted} {formatted_m}");
+
+        let formatted = {
+            let mut formatted =
+                String::from_str("  Latitude (º)\t\t\t\t  Longitude (º)\t\t\t\t  Altitude(m)\n")
+                    .unwrap();
+            let iter = division_val_x
+                .iter()
+                .zip(division_val_y.iter())
+                .zip(division_val_z.iter())
+                .zip(histogram_val_x.iter())
+                .zip(histogram_val_y.iter())
+                .zip(histogram_val_z.iter())
```

Comment:
You could consume these as they are not used after the fact. Removing the `.iter()` should work.

Reply:
Done